boards/arm/nrf5340pdk_nrf5340/nrf5340_cpunet_reset.c

```diff
@@ -51,6 +51,11 @@ static int remoteproc_mgr_boot(struct device *dev)
 	remoteproc_mgr_config();
 #endif /* !CONFIG_TRUSTED_EXECUTION_NONSECURE */
 
+#if (DT_IPC_SHM_BASE_ADDRESS != 0)
```

Comment:
Actually, @kapi-no I have one more concern, here: what is the security attribution of the shared memory? Is the shared memory secure always, or non-secure? How can we know, at this point that there is no SecureFault exception when attempting to memset the area?

Reply:
I think the shared memory region should always be non-secure since it is intended to be used by Network Core (and its non-secure image).

Reply:
No, the NETWORK MCU is not a non-secure image in all cases. In fact, it is Secure by default, IIRC. We need to guarantee this memset is not executed twice (e.g. from SPM and Non-Secure Zephyr).

 Can you make sure if we should and how we could have the following policy:
- if image is explicitly NONSECURE or, if it SECURE-by-default, do the memset, otherwise (e.g. for explicitly SECURE images, like SPM) skip it.

Btw, is this function executed by bootloader image?

Reply:
> We need to guarantee this memset is not executed twice (e.g. from SPM and Non-Secure Zephyr).
> Can you make sure if we should and how we could have the following policy:
> 
>     * if image is explicitly NONSECURE or, if it SECURE-by-default, do the memset, otherwise (e.g. for explicitly SECURE images, like SPM) skip it.
>

If this code is executed twice, then the Network Core also resets twice according to the current logic/configuration of this module. I think that shared memory contents from the previous session are invalid anyway. In my opinion, the behaviour of this module in non-secure / secure context can be topic of another PR. 

> 
> Btw, is this function executed by bootloader image?

It depends on ```BOARD_ENABLE_CPUNET``` configuration (disabled by default).

Reply:
> In my opinion, the behaviour of this module in non-secure / secure context can be topic of another PR.

It can be the topic of another PR, yes. But ultimately, you are introducing a convention here. that is, the APP image that reset the NET core has access to the shared memory. We need to document these things, cause they are currently not explicitly stated

Reply:
That's right, I added a comment highlighting this memory assumption

Reply:
Thanks. I am really not happy with the status of the code in this module, but maybe we can merge this PR now, and address the wider problem in a different PR. I will file an issue.